modules/pricer/src/main/java/com/opengamma/strata/pricer/swap/DiscountingSwapProductPricer.java

```diff
@@ -371,6 +418,18 @@ public PointSensitivityBuilder parRateSensitivity(ResolvedSwap swap, RatesProvid
    * @return the par spread curve sensitivity of the swap product
    */
   public PointSensitivityBuilder parSpreadSensitivity(ResolvedSwap swap, RatesProvider provider) {
+    // does the fixed leg of the swap, if it exists, have a future value notional
+    if (!swap.getLegs(SwapLegType.FIXED).isEmpty()) {
+      ResolvedSwapLeg fixedLeg = fixedLeg(swap);
+      SwapPaymentPeriod firstPeriod = fixedLeg.getPaymentPeriods().get(0);
+      if (firstPeriod instanceof RatePaymentPeriod) {
+        RatePaymentPeriod payment = (RatePaymentPeriod) firstPeriod;
+        RateAccrualPeriod firstAccrualPeriod = payment.getAccrualPeriods().get(0);
+        if (firstAccrualPeriod.getRateComputation() instanceof FixedOvernightCompoundedAnnualRateComputation) {
```

Comment:
Extract the logic for getting `FixedOvernightCompoundedAnnualRateComputation` to a method as it exists in multiple places in this file:

`public Optonal<FixedOvernightCompoundedAnnualRateComputation> findFixedOvernightCompoundedAnnualRateComputation();`